workflows/metaval.nf

```diff
@@ -27,14 +33,57 @@ workflow METAVAL {
     ch_versions = Channel.empty()
     ch_multiqc_files = Channel.empty()
 
+    // Create channels
+    ch_input = ch_samplesheet.branch { meta, fastq_1, fastq_2, kraken2_report, kraken2_result, kraken2_taxpasta, centrifuge_report, centrifuge_result, centrifuge_taxpasta, diamond, diamond_taxpasta ->
+
+        // Define single_end based on the conditions
+        meta.single_end = ( fastq_1 && !fastq_2 )
+
+        // reads channels
+        short_reads: meta.instrument_platform != 'OXFORD_NANOPORE'
+            return [ meta, fastq_2 ? [ fastq_1, fastq_2 ] : [ fastq_1 ] ]
+
+        long_reads: meta.instrument_platform == 'OXFORD_NANOPORE'
+            return [ meta, [ fastq_1 ] ]
+    }
+
+    // chaneels for extracting kraken2 reads
+    ch_extract_reads = ch_samplesheet.multiMap { meta, fastq_1, fastq_2, kraken2_report, kraken2_result, kraken2_taxpasta, centrifuge_report, centrifuge_result, centrifuge_taxpasta, diamond, diamond_taxpasta ->
+        meta.single_end = ( fastq_1 && !fastq_2 )
```

Comment:
Why is this defined here? We might get paired-end reads as well, right?

Reply:
Yes, this doesn't affect paired-end reads. Here, we just define the `meta.single_end` when fastq_2 is empty and which will be used down-streaming.